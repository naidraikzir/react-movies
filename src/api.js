import { get } from 'req';

export const fetchMovies = async (page, search) => {
  try {
    const { data } = await get('/', {
      params: {
        s: search,
        page: page + 1,
        type: 'movie'
      }
    });
    return data;
  } catch ({ response }) {
    return Promise.reject(response.data.Error);
  }
};

export const fetchMovie = async (id) => {
  try {
    const { data } = await get('/', {
      params: {
        i: id,
      },
    })
    return data;
  } catch ({ response }) {
    return Promise.reject(response.data.Error);
  }
};
