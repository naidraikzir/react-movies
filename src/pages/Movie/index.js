import React from 'react';

const Movie = () => (<div />);

export default Movie;
