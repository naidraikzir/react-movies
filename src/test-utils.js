import React from 'react';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { ChakraProvider, theme } from '@chakra-ui/react';
import reducer from 'store/reducer';

const customRender = (
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...options
  } = {}
) => {
  const AllProviders = ({ children }) => (
    <Provider store={store}>
      <ChakraProvider theme={theme}>{children}</ChakraProvider>
    </Provider>
  );
  return render(ui, { wrapper: AllProviders, ...options });
}

export { customRender as render };
