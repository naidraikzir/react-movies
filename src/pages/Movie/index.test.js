import React from 'react';
import { MemoryRouter } from 'react-router';
import { act } from '@testing-library/react';
import { render } from 'test-utils';
import { fetchMovie } from 'api';
import Movie from '.';

jest.mock('api');

test('match snapshot', async () => {
  fetchMovie.mockResolvedValueOnce({});
  let container;
  await act(async () => {
    const rendered = render(<MemoryRouter>
      <Movie />
    </MemoryRouter>);
    container = rendered.container;
  });
  expect(await container.firstChild).toMatchSnapshot();
});

test('renders error', async () => {
  const mockErrorMessage = 'error';
  fetchMovie.mockResolvedValueOnce({ Error: mockErrorMessage });
  let rendered;
  await act(async () => {
    rendered = render(<MemoryRouter>
      <Movie />
    </MemoryRouter>);
  });
  expect(await rendered.getByText(mockErrorMessage)).toBeInTheDocument();
});
