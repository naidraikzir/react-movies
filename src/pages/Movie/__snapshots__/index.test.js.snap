// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  class="chakra-container css-1modgg1"
>
  
  <div
    class="chakra-fade"
    style="opacity: 0;"
  >
    <div
      class="css-wklw9v"
    >
      <div
        class="css-iqgyig"
      >
        <img
          class="chakra-image css-1ypn42y"
        />
      </div>
      <div
        class="css-iqgyig"
      >
        <div
          class="css-1urwz0x"
        >
          <p
            class="chakra-text css-1ewjzqi"
          >
             
            <span
              class="css-65vpd2"
            />
          </p>
          <div
            class="css-fd0k57"
          >
            <span
              class="css-1al1bi6"
            >
              ⭐️ 
               / 
              <sub
                class="css-0"
              >
                10
              </sub>
            </span>
            <span
              class="css-m800dj"
            >
              <span
                class="css-1h8gyx6"
              />
            </span>
          </div>
          <p
            class="chakra-text css-1w2p18d"
          />
          <div
            class="css-1adflju"
          >
            <p
              class="chakra-text css-0"
            >
              Stars: 
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
